### ISSUE TITLE
XLA: random numbers are the same across `session.run` calls

### ISSUE THREAD

**Issue by yaroslavvb** [tensorflow/tensorflow#6854]
Not sure if that's intended, but that changes behavior of training pipelines:

```
from tensorflow.contrib.compiler import jit
jit_scope = jit.experimental_jit_scope
with jit_scope(compile_ops=True):
    x = tf.random_uniform(())
    
sess = tf.Session()
print(sess.run(x))
print(sess.run(x))
```

Output:
```
0.768917
0.768917
```

**Comment by yaroslavvb** [comment]
@leary for comment if that's normal

**Comment by hawkinsp** [comment]
That's definitely not working as intended. We don't seem to be setting the RNG seed correctly, which is why you get the same result.

I'll check in a change disabling JIT compilation of the RNG ops until they are fixed.

**Comment by hawkinsp** [comment]
Actually, it turns out that Tensorflow is constant-folding away the _XlaLaunch ops containing the random-number generation code. Fix coming shortly (mark _XlaLaunch as stateful).

**Comment by yaroslavvb** [comment]
Good to know it's an easy fix. BTW, if you setup the internal github/google email it'll add you to TensorFlow org automatically and I'll be able to assign future XLA issues to you (martin knows the place for the mapping)

**Comment by Leary** [comment]
Hry folks,
  Must have been some typo when you launched this thread. I am not related
to the tensor flow project. You aren't communicating with who you think you
are.

**Comment by hawkinsp** [comment]
I don't have permissions to reassign issues, but feel free to remove learyg (who is not the right Leary) and assign me.

**Comment by hawkinsp** [comment]
Ah, I see, I can't be assigned. I email Martin to add me to the Github org.

**Comment by vrv** [comment]
@hawkinsp invited

**Comment by vrv** [comment]
Also it is @leary-g I think

**Comment by yaroslavvb** [comment]
@hawkinsp btw, if you include string like this, it'll automatically close the issue when commit is merged:
"Fixes issue: #6854"
("Fixes Github issue: #6854" does not trigger it)
